src/NLog/Layouts/JsonAttribute.cs

```diff
@@ -45,17 +45,19 @@ public class JsonAttribute
         /// <summary>
         /// Initializes a new instance of the <see cref="JsonAttribute" /> class.
         /// </summary>
-        public JsonAttribute() : this(null, null) { }
+        public JsonAttribute() : this(null, null, null) { }
 
         /// <summary>
         /// Initializes a new instance of the <see cref="JsonAttribute" /> class.
         /// </summary>
         /// <param name="name">The name of the attribute.</param>
         /// <param name="layout">The layout of the attribute's value.</param>
-        public JsonAttribute(string name, Layout layout)
+        /// <param name="encode">Encode value with json-encode, true by default</param>
+        public JsonAttribute(string name, Layout layout, bool? encode)
         {
             this.Name = name;
             this.Layout = layout;
+            this.Encode = encode ?? true;
```

Comment:
Actually, none. What I wanted to do is optional parameter, but that will not binary compatible. I will change it using multiple constructors. What is the preference?